fluster/fluster.py

```diff
@@ -388,6 +392,49 @@ def _parse_suite_results(
             with open(ctx.summary_output, "w+", encoding="utf-8") as summary_file:
                 xml.write(summary_file.name, pretty=True)
 
+    def _generate_csv_summary(
+        self, ctx: Context, results: Dict[str, List[Tuple[Decoder, TestSuite]]]
+    ) -> None:
+        # pylint: disable=too-many-locals
+        result_map = {
+            TestVectorResult.SUCCESS: "Success",
+            TestVectorResult.REFERENCE: "Reference",
+            TestVectorResult.TIMEOUT: "Timeout",
+            TestVectorResult.ERROR: "Error",
+            TestVectorResult.FAIL: "Fail",
+            TestVectorResult.NOT_RUN: "Not run",
+        }
+        content = defaultdict(lambda: defaultdict(dict))
```

Comment:
Yes, I have tried that, but I find the defaultdict solution to be way shorter and cleaner. As I would otherwise have to do something like this:
```
        content = {}
        max_vectors = 0
        for test_suite, suite_results in results.items():
            if test_suite not in content:
                content[test_suite] = {}
            for decoder, vectors in suite_results:
                decoder = decoder.name[: decoder.name.find(":")]
                if decoder not in content[test_suite]:
                    content[test_suite][decoder] = {}
                max_vectors = max(max_vectors, len(vectors.test_vectors.values()))
                for vector in vectors.test_vectors.values():
                    content[test_suite][decoder][vector.name] = result_map[
                        vector.test_result
                    ]
```

Which I find ugly and unnecessary as this is exactly what defaultdicts were made for.